components/integrations/components/installed_outgoing_webhooks/installed_outgoing_webhooks.jsx

```diff
@@ -5,82 +5,108 @@ import PropTypes from 'prop-types';
-import ChannelStore from 'stores/channel_store.jsx';
-import IntegrationStore from 'stores/integration_store.jsx';
-import TeamStore from 'stores/team_store.jsx';
-import UserStore from 'stores/user_store.jsx';
-
 import * as Utils from 'utils/utils.jsx';
+import Constants from 'utils/constants.jsx';
 
 import BackstageList from 'components/backstage/components/backstage_list.jsx';
 
-import InstalledOutgoingWebhook from './installed_outgoing_webhook.jsx';
+import InstalledOutgoingWebhook from 'components/integrations/components/installed_outgoing_webhook.jsx';
+
+export default class InstalledOutgoingWebhooks extends React.PureComponent {
+    static propTypes = {
+
+        /**
+        *  Data used in passing down as props for webhook modifications
+        */
+        team: PropTypes.object,
+
+        /**
+        * Data used for checking if webhook is created by current user
+        */
+        user: PropTypes.object,
+
+        /**
+        *  Data used for checking modification privileges
+        */
+        isAdmin: PropTypes.bool,
+
+        /**
+        * Data used in passing down as props for showing webhook details
+        */
+        outgoingWebhooks: PropTypes.object,
+
+        /**
+        * Data used in sorting for displaying list and as props channel details
+        */
+        channels: PropTypes.object,
+
+        /**
+        *  Data used in passing down as props for webhook -created by label
+        */
+        users: PropTypes.object,
+
+        /**
+        *  Data used in passing as argument for loading webhooks
+        */
+        teamId: PropTypes.string,
+
+        actions: PropTypes.shape({
+
+            /**
+            * The function to call for removing outgoingWebhook
+            */
+            removeOutgoingHook: PropTypes.func,
+
+            /**
+            * The function to call for outgoingWebhook List and for the status of api
+            */
+            getOutgoingHooks: PropTypes.func,
+
+            /**
+            * The function to call for regeneration of webhook token
+            */
+            regenOutgoingHookToken: PropTypes.func
+        })
 
-export default class InstalledOutgoingWebhooks extends React.Component {
-    static get propTypes() {
-        return {
-            team: PropTypes.object,
-            user: PropTypes.object,
-            isAdmin: PropTypes.bool
-        };
     }
 
     constructor(props) {
         super(props);
 
-        this.handleIntegrationChange = this.handleIntegrationChange.bind(this);
-        this.handleUserChange = this.handleUserChange.bind(this);
         this.regenOutgoingWebhookToken = this.regenOutgoingWebhookToken.bind(this);
-        this.deleteOutgoingWebhook = this.deleteOutgoingWebhook.bind(this);
-
-        const teamId = TeamStore.getCurrentId();
+        this.removeOutgoingHook = this.removeOutgoingHook.bind(this);
+        this.outgoingWebhookCompare = this.outgoingWebhookCompare.bind(this);
```

Comment:
No change needed, but in the future you can remove these binds and use arrow syntax for the class functions (which auto bind this). A lot of the codebase still has the binds but we're trying to get rid of them as they're extra lines that can be avoided

Reply:
Ya sure. I amended the commit with these changes as well.